### ISSUE TITLE
SQLite error 21: Can't run statement that requires a customized authorizer from a cursor - while executing `PRAGMA table_xinfo`

### ISSUE THREAD

**Issue by groue** [groue/GRDB.swift#1124]
An issue was reported in the Swift forums: https://forums.swift.org/t/54371

**Comment by rpoelstra** [comment]
Hi, What exactly do you need to know additionally? The database migrator as well as the read/write logic (in ManagedDocument) is distributed in the reproduction sample (iirc). I don't think I do anything really special.
Opening a document involves a backup operation to a temporary location, from where the document is migrated.
There are no other note worthy operations on the file.

**Comment by groue** [comment]
Ah thanks, I'll check that (I was opening the provided document, and did not notice that the app was able to create this database).

**Comment by rpoelstra** [comment]
The temp URL is created at ManagedDocument:84

**Comment by groue** [comment]
All right. This gives the following minimal reproducing code:

```swift
class Observer: TransactionObserver {
    func observes(eventsOfKind eventKind: DatabaseEventKind) -> Bool { true }
    func databaseDidChange(with event: DatabaseEvent) { }
    func databaseDidCommit(_ db: Database) { }
    func databaseDidRollback(_ db: Database) { }
}
let dbQueue = DatabaseQueue()
dbQueue.add(transactionObserver: Observer(), extent: .databaseLifetime)
try dbQueue.write { db in
    try db.execute(sql: """
        CREATE VIRTUAL TABLE t1 USING rtree(id,minX,maxX,minY,maxY);
        CREATE TABLE t2 (a);
        ALTER TABLE t2 RENAME TO t3;
        """)
    _ = try db.columns(in: "t1")
}
```

It is very strange that querying the schema of t1 is impacted by the unrelated table t2 and its renaming. I believe this is worth reporting to SQLite as a bug.

**Comment by rpoelstra** [comment]
That seems very odd indeed. Will you forward this to SQLite? Or do you want me to do that?

**Comment by groue** [comment]
Thanks, but I just did it 🙂 https://sqlite.org/forum/forumpost/721da02ba2

**Comment by groue** [comment]
I found a fix: #1128
